src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -132,21 +132,23 @@ protected virtual void Dispose(bool disposing)
             OnPropertyChanged(nameof(IconBitmap));
             OnPropertyChanged(nameof(PrefixReserved));
 
-            var getVersionsTask = searchResultPackage.GetVersionsAsync();
+            Task<IReadOnlyCollection<VersionInfoContextInfo>> getVersionsTask = searchResultPackage.GetVersionsAsync();
 
             _projectVersionConstraints = new List<ProjectVersionConstraint>();
 
             // Filter out projects that are not managed by NuGet.
-            var projects = _nugetProjects.Where(project => project.ProjectKind != NuGetProjectKind.ProjectK).ToArray();
+            IProjectContextInfo[] projects = _nugetProjects.Where(project =>
+                project.ProjectKind == NuGetProjectKind.PackagesConfig || project.ProjectKind == NuGetProjectKind.PackageReference).ToArray();
```

Comment:
Was the intention to filter out `NuGetProjectKind.Unknown`?  If yes, what project produces that value?

Reply:
> Was the intention to filter out NuGetProjectKind.Unknown? 

No, my intentions are to purely refactor for efficiency.

As you can see in the logic below, we only care about projects kinds `NuGetProjectKind.PackagesConfig` and `NuGetProjectKind.PackageReference`, so passing any other projects to the service is wasteful. 

The original intention from 4+ years ago appears to be to filter out `ProjectK` types (formerly, `ProjectKNuGetProjectBase`).

> If yes, what project produces that value?

An `NuGetProjectKind.Unknown` is produced when the project is neither of type `BuildIntegratedNuGetProject` nor `MSBuildNuGetProject`.
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/ContextInfos/ProjectContextInfo.cs#L45